query_ranges_to_vnodes.cc

```diff
@@ -101,18 +101,6 @@ void query_ranges_to_vnodes_generator::process_one_range(size_t n, dht::partitio
 
         add_range(std::move(splits.first));
         cr = std::move(splits.second);
-
-        // The left bound of cr is bound({upper_bound_token, token_bound::end}, false),
-        // so we can end up with an empty range if the original end bound is its successor.
-        const bool is_empty = cr.end().has_value() &&
-            !cr.end()->is_inclusive() &&
-            !cr.end()->value().has_key() &&
-            cr.end()->value().bound() == dht::ring_position::token_bound::start &&
-            dht::token::to_int64(cr.end()->value().token()) - dht::token::to_int64(cr.start()->value().token()) == 1;
-        if (is_empty) {
-            _i++;
-            return;
-        }
```

Comment:
\cc @gusev-p

Reply:
There is a rationale in the commit message: https://github.com/scylladb/scylladb/pull/13387/commits/64674e034234ef94ed01d8676b6602a35c5a8483

Reply:
doesn't this create a regression in the series that make it less bisectable?

Reply:
I thought of the original patch as a fix, not an optimization, that just can't be triggered in the code as it is. So reverting it is dangerous.

Reply:
> I thought of the original patch as a fix, not an optimization

I think it's just an optimization. @gusev-p Is it a fix?

Reply:
> I think it's just an optimization. @gusev-p Is it a fix?

The patch just prevents data queries with empty ranges. There is no reliance on the range being not empty though, so it's more of an optimisation, yes.

Reply:
ok